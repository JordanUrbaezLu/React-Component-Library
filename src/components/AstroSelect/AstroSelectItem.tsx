import * as React from "react";

interface AstroSelectItemProps {
  children?: React.ReactNode;
}

const AstroSelectItem: React.FC<AstroSelectItemProps> = ({ children }) => {
  return <div>{children}</div>;
};

export default AstroSelectItem;
