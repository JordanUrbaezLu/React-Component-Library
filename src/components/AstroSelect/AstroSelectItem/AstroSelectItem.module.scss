@use "../../../utility/globals";

.astroSelectItem {
  font-family: Verdana, Geneva, Tahoma, sans-serif;
  text-align: left;
  padding: 5px 5px 5px 12px;
  color: white;
  background-color: rgb(50, 50, 50);
  cursor: pointer;

  &:hover {
    background-color: rgb(80, 80, 80);
  }

  &.small {
    font-size: 10px;
  }

  &.medium {
    font-size: 14px;
  }

  &.large {
    font-size: 18px;
  }
}
