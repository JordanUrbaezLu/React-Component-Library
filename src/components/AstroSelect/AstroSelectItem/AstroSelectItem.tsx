import * as React from "react";
interface AstroSelectItemProps {
  value: string;
}

const AstroSelectItem: React.FC<AstroSelectItemProps> = ({ value }) => {
  return <span>{value}</span>;
};

export default AstroSelectItem;
