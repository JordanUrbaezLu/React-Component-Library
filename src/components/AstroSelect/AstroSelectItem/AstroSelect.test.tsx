import { render, screen } from "@testing-library/react";
import AstroSelectItem from "./AstroSelectItem";
import { axe, toHaveNoViolations } from "jest-axe";
import "@testing-library/jest-dom";
expect.extend(toHaveNoViolations);

describe("AstroSelect", () => {
  test("Should render component correctly.", () => {
    const { container } = render(<AstroSelectItem value="item" />);
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe("Accessibility", () => {
  test("Should have no accessibility violations.", async () => {
    const { container } = render(<AstroSelectItem value="item" />);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });
});

describe("Props", () => {
  test("Should render value correctly.", () => {
    render(
      <div data-testid="select">
        <AstroSelectItem value="item" />
      </div>
    );
    expect(screen.getByTestId("select")).toHaveTextContent("item");
  });
});
