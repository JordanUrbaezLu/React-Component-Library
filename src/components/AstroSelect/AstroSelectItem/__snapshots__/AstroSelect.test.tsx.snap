// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AstroSelect Should render component correctly. 1`] = `
<span>
  item
</span>
`;
